### PR TITLE
Implement defaultargument tests in Chromium

### PR DESCRIPTION
**Pull request by kblok**
progress on #68

### CODE REVIEW

File: src/PlaywrightSharp.Abstractions/ITarget.cs [hunk 1]
```diff
@@ -9,5 +9,12 @@ public interface ITarget
         /// Gets the URL.
         /// </summary>
         string Url { get; }
+
+        /// <summary>
+        /// Gets the type. It will be <see cref="ITarget.Type"/>.
+        /// Can be `"page"`, `"background_page"`, `"service_worker"`, `"shared_worker"`, `"browser"` or `"other"`.
```
Comment (Meir017) [inline src/PlaywrightSharp.Abstractions/ITarget.cs]: no need to specify this. The enum has these values...

**Comment by kblok** [comment]
I have some reds but it's ready to go @Meir017 !

File: src/PlaywrightSharp.Abstractions/IBrowserFetcher.cs [hunk 2]
```diff
@@ -15,11 +15,11 @@ public interface IBrowserFetcher
         public event DownloadProgressChangedEventHandler DownloadProgressChanged;
 
         /// <summary>
-        /// Gets the revision info.
+        /// Gets the revision info. It will return the default revision info if <paramref name="revision"/> is not set.
```
Comment (Meir017) [inline src/PlaywrightSharp.Abstractions/IBrowserFetcher.cs]: consider moving this to a `remarks` comment

File: src/PlaywrightSharp.Abstractions/TargetType.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        /// </summary>
+        Javascript,
+
+        /// <summary>
+        /// Target type network
+        /// </summary>
+        Network,
+
+        /// <summary>
+        /// Target type network
```
Comment (Meir017) [inline src/PlaywrightSharp.Abstractions/TargetType.cs]: this doesn't seem right

File: src/PlaywrightSharp.Chromium/ChromiumBrowserType.cs [hunk 4]
```diff
@@ -88,19 +123,176 @@ public IBrowserFetcher CreateBrowserFetcher(BrowserFetcherOptions options = null
...
+            if (customEnv != null)
+            {
+                foreach (var item in customEnv)
+                {
+                    environment[item.Key] = item.Value;
+                }
+            }
+        }
+
+        private string GeChromeExecutablePath(LaunchOptions options)
```
Comment (Meir017) [inline src/PlaywrightSharp.Chromium/ChromiumBrowserType.cs]: rename to `GetChromeExecutablePath`

File: src/PlaywrightSharp.Chromium/ChromiumBrowserType.cs [hunk 5]
```diff
@@ -88,19 +123,176 @@ public IBrowserFetcher CreateBrowserFetcher(BrowserFetcherOptions options = null
...
+                    {
+                        return;
+                    }
+
+                    var transport = await BrowserHelper.CreateTransportAsync(browserApp.GetConnectOptions()).ConfigureAwait(false);
+                    await transport.SendAsync("Browser.close", new BrowserCloseRequest { Id = BrowserCloseMessageId }).ConfigureAwait(false);
+                },
+                (exitCode) =>
+                {
+                    if (browserApp != null)
```
Comment (Meir017) [inline src/PlaywrightSharp.Chromium/ChromiumBrowserType.cs]: you could just `browserApp?.ProcessKilled(exitCode);`

File: src/PlaywrightSharp.Chromium/Messaging/Browser/BrowserCloseRequest.cs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+namespace PlaywrightSharp.Chromium.Messaging.Browser
```
Comment (Meir017) [inline src/PlaywrightSharp.Chromium/Messaging/Browser/BrowserCloseRequest.cs]: we should consider generating the protocol classes in a similar way to playwright - [utils\protocol-types-generator\index.js](https://github.com/kblok/playwright/blob/v0.10.0/utils/protocol-types-generator/index.js)

Comment (kblok) [reply src/PlaywrightSharp.Chromium/Messaging/Browser/BrowserCloseRequest.cs]: Issue created! https://github.com/kblok/playwright-sharp/issues/229

Feel free to grab it. I know you like those tools :)

**Pull request merged**